### PR TITLE
added scheduler

### PR DESCRIPTION
**Pull request by venkibric**
### JIRA link (if applicable) ###

https://crowncommercialservice.atlassian.net/browse/SCAT-4840

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

### CODE REVIEW

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/config/BuyerUserScheduler.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  @Value("${spring.security.oauth2.client.provider.jaggaer.token-uri}")
+  private String tokenUri;
+
+  private final BuyerUserDetailsRepo buyerDetailsRepo;
+  private final JaggaerAPIConfig jaggaerAPIConfig;
+  private XSSFWorkbook workbook;
+  private XSSFSheet sheet;
+  private static final String SHEET_NAME = "Buyers_Details";
+
+  @Scheduled(fixedDelayString = "PT1M")
```
Comment (venkibric) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/config/BuyerUserScheduler.java:54]: Will change the scheduler time

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/config/BuyerUserScheduler.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    usersList.stream().forEach(user -> {
+      int rowCount = 1;
+      Row row = sheet.createRow(rowCount++);
+      int columnCount = 0;
+      createCell(row, columnCount++, user.getUserId(), style);
+      createCell(row, columnCount++, user.getUserPassword(), style);
+    });
+  }
+
+  private GetCompanyDataResponse getSelfServiceBuyers() {
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/config/BuyerUserScheduler.java:127]: This is a bit troubling - we shouldn't have to reconfigure the Jaggaer web client.  I know you mentioned an auth issue with it, would be good to revisit when we can and figure out what's going on.  I shall try to have a look 👍

Comment (tommyb82) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/config/BuyerUserScheduler.java:127]: @venkibric - did you have a chance to revisit this?  There is some specific config for the Jaggaer web client that we don't want to miss or duplicate - if we could debug and find out why the normal autowired client was not working that would be great 👍

**Comment by venkibric** [comment]
I will revisit again this night and we discuss tomorrow morning in this

Regards,
Venki
________________________________

**Comment by tommyb82** [comment]
Closing this PR - will start new branch with scheduler on for S3 export (SCAT-4842) @venkibric FYI

**Pull request closed without merging**